pkg/util/ratelimiter/ratelimiter_test.go

```diff
@@ -33,28 +52,25 @@ func TestRateLimitedFunction(t *testing.T) {
 	}
 
 	for _, tc := range tests {
-		counter := 0
-		handler := func() error {
-			counter += 1
-			return nil
-		}
-
+		h := &handler{}
 		quit := make(chan struct{})
-		rlf := NewRateLimitedFunction(keyFunc, tc.Interval, handler)
+		rlf := NewRateLimitedFunction(keyFunc, tc.Interval, h.handle)
 		rlf.RunUntil(quit)
 
 		for i := 0; i < tc.Times; i++ {
-			go func() {
-				if tc.Interval > 0 {
+			go func(rlf *RateLimitedFunction, idx, interval int) {
+				if interval > 0 {
 					rlf.Invoke(rlf)
 				} else {
-					rlf.Invoke(i)
+					rlf.Invoke(idx)
 				}
-			}()
+			}(rlf, i, tc.Interval)
```

Comment:
Don't really need this as what we are only interested in is that Invoke is called - the parameter is not really used specifically for this test. But its ok to put this change in.